service/src/main/java/bio/terra/cbas/dependencies/sam/SamService.java

```diff
@@ -27,10 +44,42 @@ private StatusApi getStatusApi() {
     return new StatusApi(samClient.getApiClient());
   }
 
+  boolean hasPermission(String actionType) {
+    // don't check auth access with Sam if "sam.checkAuthAccess" is false
+    if (!samClient.checkAuthAccessWithSam()) return true;
+
+    logger.debug(
+        "Checking Sam permission for '{}' resource and '{}' action type for user on workspace '{}'.",
+        RESOURCE_TYPE_WORKSPACE,
+        actionType,
+        samClient.getWorkspaceId());
+
+    try {
+      ResourcesApi resourcesApi = getResourcesApi();
+      return SamRetry.retry(
+          () ->
+              resourcesApi.resourcePermissionV2(
+                  RESOURCE_TYPE_WORKSPACE, samClient.getWorkspaceId(), actionType));
+    } catch (ApiException e) {
+      throw SamExceptionFactory.create(
+          "Error checking %s permissions on workspace from Sam".formatted(actionType), e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw SamExceptionFactory.create(
+          "Request interrupted while checking %s permissions on workspace from Sam"
+              .formatted(actionType),
+          e);
+    }
```

Comment:
This looks very similar to the `getUserStatusInfo` method body. Maybe outside of the scope of this PR but it could be nice to abstract out the similarities if not too complex.

Reply:
Yeah that is true. Abstracting out the try block and different error messages passed to `SamExceptionFactory.create` makes sense but I am not sure how it will affect the readability of the code. 🤔 If you have strong preference I can try generalizing it in this PR?